src/core/shape_inference/include/deformable_convolution_shape_inference.hpp

```diff
@@ -18,7 +18,14 @@ size_t calculate_num_spatial(const util::DeformableConvolutionBase* op, const st
     auto num_spatial = util::num_spatial_from_shapes(input_shapes[0], input_shapes[2], non_spatial_count);
 
     if (num_spatial == convolution::num_spatial_undefined && input_shapes[1].rank().is_static()) {
-        num_spatial = input_shapes[1].size() - non_spatial_count;
+        const auto offsets_rank = input_shapes[1].size();
+        NODE_VALIDATION_CHECK(op,
+                              offsets_rank >= non_spatial_count,
```

Comment:
This check should not be required as inputs rank should be validated before call this helper (maybe there is bug and input is not validation). The rank of `inputs[1]` is `4` when static, this condition should be always true.

If this assert is required add test to catch it (or check if exists already).

Reply:
Replaced with fixed offsets rank when static.